jupyterhub/app.py

```diff
@@ -1966,25 +1965,27 @@ async def init_groups(self):
                 group.users.append(user)
         db.commit()
 
-    async def init_roles(self):
+    async def init_role_creation(self):
         """Load default and predefined roles into the database"""
-        db = self.db
-        # tokens are added separately
-        role_bearers = ['users', 'services', 'groups']
-
-        # load default roles
         self.log.debug('Loading default roles to database')
         default_roles = roles.get_default_roles()
         for role in default_roles:
-            roles.create_role(db, role)
+            roles.create_role(self.db, role)
 
+    async def init_role_assignment(self):
+        # tokens are added separately
+        role_bearers = ['users', 'services', 'groups']
+
+        db = self.db
         # load predefined roles from config file
         self.log.debug('Loading predefined roles from config file to database')
         for predef_role in self.load_roles:
             roles.create_role(db, predef_role)
```

Comment:
`create_role` should be in role_creation, right?

Reply:
Yes, also agree, role_creation should create the custom roles as well, not just default.

Reply:
Good point, moved it!